### PR TITLE
[DreamBooth] add text encoder LoRA support in the DreamBooth training script

### PR DESCRIPTION
**Pull request by sayakpaul**
Following up on https://github.com/huggingface/diffusers/pull/2918. Closes https://github.com/huggingface/diffusers/issues/3065

Hyperparameter tuning might change the game but with the default hyperparameters (taken from the LoRA training section of DreamBooth), it seems to be doing okay. 

Report: https://wandb.ai/sayakpaul/dreambooth-lora/reports/test-23-04-17-17-00-13---Vmlldzo0MDkwNjMy

Model repo (comes with a card if someone forgot): https://huggingface.co/sayakpaul/dreambooth

Sharing this cutie which was generated:
image

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Great! Could we maybe also update the docs:
https://github.com/huggingface/diffusers/blob/main/docs/source/en/training/lora.mdx
and tests:
https://github.com/huggingface/diffusers/blob/main/examples/test_examples.py (think we've never added a lora test - my bad I added the dreambooth lora training script :sweat_smile: )

File: examples/dreambooth/train_dreambooth_lora.py [hunk 1]
```diff
@@ -900,7 +948,11 @@ def main(args):
 
                 accelerator.backward(loss)
                 if accelerator.sync_gradients:
-                    params_to_clip = lora_layers.parameters()
+                    params_to_clip = (
+                        itertools.chain(unet_lora_layers.parameters(), text_encoder_lora_layers.parameters())
+                        if args.train_text_encoder
+                        else text_encoder_lora_layers.parameters()
```
Comment (themrzmaster) [inline examples/dreambooth/train_dreambooth_lora.py:954]: shouldn't it be 
`else text_encoder_lora_layers.parameters()`
?

**Comment by sayakpaul** [comment]
@patrickvonplaten ready for a review now. I have addressed your comments. 

The tests were run using the command below:

```bash
python -m pytest -n 2 --max-worker-restart=0 --dist=loadfile -s -v -k "lora" examples/test_examples.py
```

File: docs/source/en/training/lora.mdx [hunk 2]
```diff
@@ -175,6 +177,11 @@ accelerate launch train_dreambooth_lora.py \
   --push_to_hub
 ```
 
+It's also possible to additionally fine-tune the text encoder with LoRA. This, in most cases, leads
+to better results with a slight increase in the compute. To allow fine-tuning the text encoder with LoRA,
```
Comment (innat-asj) [inline docs/source/en/training/lora.mdx:181]: Would it be possible to add more experimental details of `unet_lora`  and `text_encoder_lora` (and combined) in separate blogs (like [dreambooth](https://huggingface.co/blog/dreambooth))?

Comment (sayakpaul) [reply docs/source/en/training/lora.mdx:181]: We are leaving it to the community for now. But we will update if we find more :)

**Review by patrickvonplaten — APPROVED** [review]
Great job! Good to merge for me - we'll just need to resolve the merge conflicts

**Comment by tcapelle** [comment]
Reverting to previous commits fix the issue

**Comment by sayakpaul** [comment]
It's better if you create a new issue thread for this problems you are facing with a reproducible snippet.

**Comment by tcapelle** [comment]
Yeah sorry, but I just pulled and integrated that commit and it broke, and realised that you just had merged that.

**Pull request merged**